d4s2_api_v2/serializers.py

```diff
@@ -54,3 +55,71 @@ class Meta:
     @staticmethod
     def get_duke_ds_user(user):
         return DDSUtil(user).get_current_user().id
+
+
+class S3EndpointSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = S3Endpoint
+        resource_name = 's3endpoints'
+        fields = ('id', 'url',)
+
+
+class S3UserSerializer(serializers.ModelSerializer):
+    email = serializers.SerializerMethodField()
+    type = serializers.SerializerMethodField()
+
+    class Meta:
+        model = S3User
+        resource_name = 's3users'
+        fields = ('id', 'user', 'endpoint', 'email', 'type')
+
+    @staticmethod
+    def get_email(s3_user):
+        return s3_user.user.email
+
+    @staticmethod
+    def get_type(s3_user):
+        return s3_user.get_type_label()
+
+
+class S3BucketSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = S3Bucket
+        resource_name = 's3bucket'
+        fields = ('id', 'name', 'owner', 'endpoint')
+
+    def validate_owner(self, owner):
```

Comment:
Might be worth a comment that this `validate_<fieldname>` is automatically picked up and run by the serializer (I wasn't familiar with this behavior and looked for an explicit call)

Reply:
I added a comment to this effect.